common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -676,6 +703,42 @@ class @Problem
       element = $(element)
       element.find("section[id^='forinput']").removeClass('choicetextgroup_show_correct')
 
+  enableHintButton: (enable) =>
+    # Used to disable/enable hint button.
+    if enable
+      @hintButton.removeClass 'is-disabled'
+      @hintButton.attr({'aria-disabled': 'false'})
+    else
+      @hintButton.addClass 'is-disabled'
+      @hintButton.attr({'aria-disabled': 'true'})
+
+  enableResetButton: (enable) =>
+    # Used to disable/enable reset button.
+    if enable
+      @resetButton.removeClass 'is-disabled'
+      @resetButton.attr({'aria-disabled': 'false'})
+    else
+      @resetButton.addClass 'is-disabled'
+      @checkButton.attr({'aria-disabled': 'true'})
+
+  enableSaveButton: (enable) =>
+    # Used to disable/enable save button.
+    if enable
+      @saveButton.removeClass 'is-disabled'
+      @saveButton.attr({'aria-disabled': 'false'})
+    else
+      @saveButton.addClass 'is-disabled'
+      @saveButton.attr({'aria-disabled': 'true'})
+
+  enableShowButton: (enable) =>
+    # Used to disable/enable show answer button.
+    if enable
+      @showButton.removeClass 'is-disabled'
+      @showButton.attr({'aria-disabled': 'false'})
+    else
+      @showButton.addClass 'is-disabled'
+      @showButton.attr({'aria-disabled': 'true'})
+
```

Comment:
Completely agree. Think about other ways you can make this code more DRY. Can you make a function, `disableButtonsWhileRunning`, that takes functions like `save_internal` or `reset` and disables all buttons while they're running?

Another thing to look out for: `enableCheckButton` changes the text of the check button while it's being clicked. We only want that when it's the check button that has been clicked. If you click "save", you don't want the check button's text to be changed, but you still want the check button to be disabled.